CHANGELOG.md

```diff
@@ -1,3 +1,15 @@
+## 2.31.0 (June 4th, 2025)
+
+[Full Changelog](https://github.com/confluentinc/terraform-provider-confluent/compare/v2.30.0...v2.31.0)
+
+**New features:**
+* Added `confluent_flink_connection` resource and data source to manage connections between Flink statements and UDFs and third party services.
+* Added `confluent_connect_artifact` resource and data source to manage Custom SMTs (Single Message Transform) for fully managed connectors.
```

Comment:
Let's add links to the resource and data source docs and mention the lifecycle stage for these.

Some examples are the changelog notes for `confluent_tableflow_topic` or `confluent_flink_statement`